runtime/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use rstd::{vec::Vec, collections::btree_map::BTreeMap};
+use codec::{Encode, Decode};
+use runtime_primitives::traits::Block as BlockT;
+
+#[cfg(not(feature = "std"))]
+#[doc(hidden)]
+pub use rstd::slice;
+
+#[macro_use]
+pub mod validate_block;
+
+type WitnessData = BTreeMap<Vec<u8>, Vec<u8>>;
```

Comment:
Surely a hashmap is faster?

Reply:
Yeah can switch to a hashmap, just will require some external hashmap, as std hashmap does not work on `no_std`. AFAIK.

Reply:
`hashbrown` and a `no-std` wrapper implementing `Decode` I guess. But let's get to that later